### DISCUSSION TITLE
Enhancement Request: Use "gaps" to search for missing movies

### DISCUSSION THREAD

**Discussion by I-am-PUID-0** [itsToggle/plex_debrid#258] [Ideas]
Hey @itsToggle, 

You have done a phenomenal job! 

I know it's a big ask, but is there any way to incorporate gathering new content from gaps, which is another project that scans your existing movie library for missing moves of the same collection? It provides an RSS feed as an output. 

https://github.com/JasonHHouse/gaps

THX!

**Comment by itsToggle** [comment]
Hey! Thanks man, I will look into it 👍 Ive created a feature request issue from this, so I dont forget :)

**Reply by I-am-PUID-0** [reply]
Awesome! Thanks so much!

**Reply by itsToggle** [reply]
@I-am-PUID-0 hey, I think Ive got some time to look into this. Could you send me some example xml file that the rss feed returns (without private information ofc) ? :) That way I wont have to actually set up gaps myself.

**Reply by I-am-PUID-0** [reply]
@itsToggle, 

Hey, thanks for following up. Unfortunately, I think GAPS is a dead program, and I've been having issues with it lately. 

However, I did also have another thought about a similar capability/feature that would leverage Sonarr v3 or Radarr v3 and likely provide more utility to a more extensive user set. 

The basic premise would be to get the missing/wanted data from Sonarr/Radarr after the user "imports and monitors" their debrid library linked to plex_debrid -- or any media library. 

The below will provide the output from Sonarr of any missing "monitored" content. Sonarr API docs: https://sonarr.tv/docs/api/

`curl localhost:8989/api/wanted/missing?pageSize=1000 -X GET --header "X-Api-Key:<api-key>"`

Here's an example of the output utilizing the {id} tag for the API. e.g. `curl localhost:8989/api/wanted/missing/{1}?pageSize=1000 -X GET --header "X-Api-Key:<api-key>"`

`{
  "seriesId": 2,
  "episodeFileId": 12,
  "seasonNumber": 1,
  "episodeNumber": 1,
  "title": "The Bad Beginning: Part One",
  "airDate": "2017-01-13",
  "airDateUtc": "2017-01-13T08:00:00Z",
  "overview": "The dreadful history of the Baudelaire children begins with a deadpan narrator, a terrible fire and the ominous arrival of a distant relative.",
  "episodeFile": {
    "seriesId": 2,
    "seasonNumber": 1,
    "relativePath": "Season 01/A Series of Unfortunate Events (2017) - S01E01 - The Bad Beginning Part One [WEBRip-1080p][AC3 5.1][x264]-DEFLATE.mkv",
    "path": "/media/shows/A Series of Unfortunate Events (2017) [imdb-tt4834206]/Season 01/A Series of Unfortunate Events (2017) - S01E01 - The Bad Beginning Part One [WEBRip-1080p][AC3 5.1][x264]-DEFLATE.mkv",
    "size": 4925583553,
    "dateAdded": "2023-01-05T21:29:50.74793Z",
    "quality": {
      "quality": {
        "id": 15,
        "name": "WEBRip-1080p",
        "source": "webRip",
        "resolution": 1080
      },
      "revision": {
        "version": 1,
        "real": 0,
        "isRepack": false
      }
    },
    "language": {
      "id": 1,
      "name": "English"
    },
    "mediaInfo": {
      "audioChannels": 5.1,
      "audioCodec": "AC3",
      "videoCodec": "x264"
    },
    "qualityCutoffNotMet": true,
    "id": 12
  },
  "hasFile": true,
  "monitored": true,
  "absoluteEpisodeNumber": 1,
  "unverifiedSceneNumbering": false,
  "series": {
    "title": "A Series of Unfortunate Events",
    "sortTitle": "series of unfortunate events",
    "seasonCount": 3,
    "status": "ended",
    "overview": "The extraordinary Baudelaire orphans face trials, tribulations and the evil Count Olaf in their fateful quest to unlock long-held family secrets.",
    "network": "Netflix",
    "airTime": "03:00",
    "images": [
      {
        "coverType": "banner",
        "url": "https://artworks.thetvdb.com/banners/graphical/306304-g4.jpg"
      },
      {
        "coverType": "poster",
        "url": "https://artworks.thetvdb.com/banners/posters/306304-6.jpg"
      },
      {
        "coverType": "fanart",
        "url": "https://artworks.thetvdb.com/banners/fanart/original/306304-2.jpg"
      }
    ],
    "seasons": [
      {
        "seasonNumber": 1,
        "monitored": true
      },
      {
        "seasonNumber": 2,
        "monitored": true
      },
      {
        "seasonNumber": 3,
        "monitored": true
      }
    ],
    "year": 2017,
    "path": "/media/shows/A Series of Unfortunate Events (2017) [imdb-tt4834206]",
    "profileId": 6,
    "languageProfileId": 1,
    "seasonFolder": true,
    "monitored": true,
    "useSceneNumbering": false,
    "runtime": 47,
    "tvdbId": 306304,
    "tvRageId": 0,
    "tvMazeId": 3949,
    "firstAired": "2017-01-13T00:00:00Z",
    "lastInfoSync": "2023-01-05T22:45:38.685416Z",
    "seriesType": "standard",
    "cleanTitle": "aseriesunfortunateevents",
    "imdbId": "tt4834206",
    "titleSlug": "a-series-of-unfortunate-events",
    "certification": "TV-PG",
    "genres": [
      "Adventure",
      "Children",
      "Comedy",
      "Family"
    ],
    "tags": [],
    "added": "2023-01-05T21:27:44.854092Z",
    "ratings": {
      "votes": 0,
      "value": 0.0
    },
    "qualityProfileId": 6,
    "id": 2
  },
  "id": 1`



Radar also has an API call for collections: `curl localhost:7878/api/v3/collection/{1}?pageSize=1000 -X GET --header "X-Api-Key:<api-key>"`

Radarr API docs: https://radarr.video/docs/api/


Here's an example of the output utilizing the {id} tag for the API. e.g. `curl localhost:7878/api/v3/collection/{1}?pageSize=1000 -X GET --header "X-Api-Key:<api-key>"`

`
{
  "title": "American Psycho Collection",
  "sortTitle": "american psycho collection",
  "tmdbId": 86105,
  "images": [
    {
      "coverType": "poster",
      "url": "https://image.tmdb.org/t/p/original/ko4O9lcRGYOZAF5Hh79Mfsgjpun.jpg"
    },
    {
      "coverType": "fanart",
      "url": "https://image.tmdb.org/t/p/original/z5cYtp6jrS23RrQPBiJMJpIMwjc.jpg"
    }
  ],
  "overview": "American Psycho is a cult psychological thriller and satirical film. The film focuses on Wall Street yuppie Patrick Bateman, whose mental instability and bloodlust lead him to serial killing. A direct-to-video spin-off, American Psycho II: All American Girl, was not based on the novel or the original film, as its only connection with the original is the death of Patrick Bateman, briefly shown in a flashback. However, the sequel continues the pattern in the first film of featuring a series of graphic deaths of colleagues and friends of the main character.",
  "monitored": true,
  "rootFolderPath": "/media/movies",
  "qualityProfileId": 6,
  "searchOnAdd": false,
  "minimumAvailability": "released",
  "movies": [
    {
      "tmdbId": 1359,
      "imdbId": "tt0144084",
      "title": "American Psycho",
      "cleanTitle": "americanpsycho",
      "sortTitle": "american psycho",
      "overview": "A wealthy New York investment banking executive hides his alternate psychopathic ego from his co-workers and friends as he escalates deeper into his illogical, gratuitous fantasies.",
      "runtime": 102,
      "images": [
        {
          "coverType": "poster",
          "url": "https://image.tmdb.org/t/p/original/9uGHEgsiUXjCNq8wdq4r49YL8A1.jpg"
        },
        {
          "coverType": "fanart",
          "url": "https://image.tmdb.org/t/p/original/bneUTWFAcVCdsb0O5UwZbJd8xqZ.jpg"
        }
      ],
      "year": 2000,
      "ratings": {
        "imdb": {
          "votes": 616848,
          "value": 7.6,
          "type": "user"
        },
        "tmdb": {
          "votes": 8672,
          "value": 7.4,
          "type": "user"
        },
        "metacritic": {
          "votes": 0,
          "value": 64,
          "type": "user"
        },
        "rottenTomatoes": {
          "votes": 0,
          "value": 68,
          "type": "user"
        }
      },
      "genres": [
        "Thriller",
        "Drama",
        "Crime"
      ],
      "folder": "American Psycho (2000) {imdb-tt0144084}"
    },
    {
      "tmdbId": 10726,
      "imdbId": "tt0283877",
      "title": "American Psycho II: All American Girl",
      "cleanTitle": "americanpsychoiiallamericangirl",
      "sortTitle": "american psycho ii all american girl",
      "overview": "Rachel is a criminology student hoping to land a position as a teacher's assistant for professor Robert Starkman. She's sure this position will pave the way to an FBI career, and she's willing to do anything to obtain it -- including killing her classmates. The school psychiatrist, Dr. Daniels, becomes aware that Rachel is insane, but Rachel is skilled at her dangerous game of death and identity theft.",
      "runtime": 88,
      "images": [
        {
          "coverType": "poster",
          "url": "https://image.tmdb.org/t/p/original/mUkuaPoC21Dtemnx583THBEdHG0.jpg"
        },
        {
          "coverType": "fanart",
          "url": "https://image.tmdb.org/t/p/original/bck1APsiyOtEniGLsFO0Gu8QtUU.jpg"
        }
      ],
      "year": 2002,
      "ratings": {
        "imdb": {
          "votes": 16637,
          "value": 3.7,
          "type": "user"
        },
        "tmdb": {
          "votes": 318,
          "value": 4.6,
          "type": "user"
        },
        "rottenTomatoes": {
          "votes": 0,
          "value": 0,
          "type": "user"
        }
      },
      "genres": [
        "Horror",
        "Thriller"
      ],
      "folder": "American Psycho II All American Girl (2002) {imdb-tt0283877}"
    }
  ],
  "id": 1
`




As an aside, depending upon how much integration is available from Sonarr's or Radarr's API, either could ostensibly serve as a UI replacement for plex_debrid; However, I'm sure it would be a significant undertaking.
